Dark and Light Theme Toggle Functionality

stevenovak123 opened the pull request.

stevenovak123 commented:
I ran the scripts in the frontend to lint and still they have shown up in the changes. Sorry for the difficulty caused.

nirsht commented:
@stevenovak123 can you explain more in details what is the problem ?

nirsht commented:
@stevenovak123 Can you also attach pics to here ?

nirsht commented on frontend/src/App.tsx:
```diff
@@ -1,33 +1,63 @@
-        <GlobalStyle />
-        <PrivateRoutes />
-      </Container>
+      <ThemePreference.Provider value={{ currentTheme, setCurrentTheme }}>
```
I rather seperate it to different file containing thr ThemeProvider

stevenovak123 replied:
I tried separating it and ran into many type errors and problems.

nirsht commented on frontend/src/components/MainView/MainViewHeader.tsx:
```diff
@@ -91,43 +91,51 @@ const LogoValidkubeMobileContainer = styled(LogoModblieValidkube)`
+  const theme = useContext(ThemePreference);
+
+  const togglerTheme = theme?.currentTheme === 'dark' ? 'light' : 'dark';
+  console.log(togglerTheme);
```
Please remove that

nirsht reviewed (commented):
Also please try to solve the changes from `""` to `''`, this is probably the formatter that you are using, try use prettier

stevenovak123 commented:
> @stevenovak123 can you explain more in details what is the problem ?

I use prettier as well but my config has them in ' ' instead of " ". I think it is due to the fact there is not a prettier rules file set in the project. if that is that it will automatically lint it to a default set.

stevenovak123 commented:
Added the necessary changes as requested. separated the context logic and corrected the " ". Attaching the images of how the changes look. The theme switching button is located on the Navbar beside the github icon.
![validkube](https://user-images.githubusercontent.com/59828288/192031684-46506901-f2b1-4bb8-979f-a7eff7d7940a.png)
![validkubeLightMode_](https://user-images.githubusercontent.com/59828288/192031697-98ac7e25-4a79-4765-86bb-3856a8b91aab.png)

nirsht commented:
Hi @stevenovak123 looks great, just to verify, the default theme is dark ?

I am AFK for few days, after that I will merge it and deploy it 🎉

stevenovak123 commented:
> Hi @stevenovak123 looks great, just to verify, the default theme is dark ?
> 
> I am AFK for few days, after that I will merge it and deploy it 🎉

The default is set depending on the users' browser settings.
